### PR TITLE
Support for DotNet 5.0 preview

### PR DESCRIPTION
**Pull request by purva-vasudeo**
Changes include new builders for dotnet 5.0 and code to detect if target framework of the project is 5.0

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/projectkudu/kudu?pullRequest=3192) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/projectkudu/kudu?pullRequest=3192) before we can accept your contribution.<br/><br/>:x: purva-vasudeo [sign now](https://cla.dotnetfoundation.org/projectkudu/kudu?pullRequest=3192)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/projectkudu/kudu?pullRequest=3192) it.</sub>

File: Kudu.Core/Deployment/Generator/SiteBuilderFactory.cs [hunk 1]
```diff
@@ -383,7 +421,16 @@ private ISiteBuilder DetermineProject(string repositoryRoot, string targetPath,
             }
             else if (AspNetCoreHelper.IsDotnetCoreFromProjectFile(targetPath, projectTypeGuids))
             {
-                if (VsHelper.UseMSBuild16())
+                if (VsHelper.IsDotNetCore5(targetFramework)){
```
Comment (suwatch) [inline Kudu.Core/Deployment/Generator/SiteBuilderFactory.cs:424]: indent

File: Kudu.Core/Deployment/Generator/SiteBuilderFactory.cs [hunk 2]
```diff
@@ -428,7 +484,16 @@ private ISiteBuilder DetermineProject(string repositoryRoot, string targetPath,
             {
                 if (FunctionAppHelper.IsCSharpFunctionFromProjectFile(targetPath))
                 {
-                    if (VsHelper.UseMSBuild16())
+                    if (VsHelper.IsDotNetCore5(targetFramework))
```
Comment (suwatch) [inline Kudu.Core/Deployment/Generator/SiteBuilderFactory.cs:487]: VsHelper.UseMSBuild1607

File: Kudu.Core/Deployment/Generator/SiteBuilderFactory.cs [hunk 3]
```diff
@@ -404,7 +451,16 @@ private ISiteBuilder DetermineProject(string repositoryRoot, string targetPath,
             }
             else if (VsHelper.IsExecutableProject(targetPath))
             {
-                if (VsHelper.UseMSBuild16())
+                if (VsHelper.IsDotNetCore5(targetFramework))
+                {
+                    return new DotNetCoreConsoleNewMSBuildBuilder(_environment,
```
Comment (suwatch) [inline Kudu.Core/Deployment/Generator/SiteBuilderFactory.cs:456]: MSBuild1607

**Pull request merged**